src/java/org/apache/cassandra/index/sai/utils/RangeIterator.java

```diff
@@ -244,16 +237,13 @@ else if (tokenCount > 0 && max.compareTo(range.getMinimum()) > 0)
                             tokenCount = Math.min(tokenCount, range.getMaxKeys());
                         else
                             tokenCount = range.getMaxKeys();
+
                         break;
 
                     default:
                         throw new IllegalStateException("Unknown iterator type: " + iteratorType);
                 }
 
-                // check if new range is disjoint with already added ranges, which means that this intersection
-                // is not going to produce any results, so we can cleanup range storage and never added anything to it.
-                isOverlapping &= isOverlapping(min, max, range);
```

Comment:
Looks like we don't need the `isOverlapping` method any more as it is only used in tests. Is there any reason we should keep the method?

Reply:
I decided to update the tests and remove the method.